plugins/pluginmanagers/lazy.nix

```diff
@@ -41,10 +41,13 @@ in
     plugins.lazy = {
       enable = mkEnableOption "lazy.nvim";
 
+      setup = mkOption { type = types.submodule { freeformType = types.attrsOf types.anything; }; };
```

Comment:
You probably want `mkSettingsOption`: https://github.com/nix-community/nixvim/blob/665680a5caeb57ff73e0a112ac805f5456f2a91d/lib/options.nix#L337-L342

Although, if there's no sub-options declared, we could also just make this a `type = attrsOf anything` option without using a submodule. (I'd prefer this)

Either way, we should use the standard description _"Options provided to the `require('lazy').setup` function."_ https://github.com/nix-community/nixvim/blob/665680a5caeb57ff73e0a112ac805f5456f2a91d/lib/neovim-plugin.nix#L35

We'd also want to name the option `settings`. While I agree naming it `setup` does make sense, we've standardized on using `settings` for structured config that'll be passed to a plugins configuration method (i.e. usually the `setup` function).

On reflection, I think it makes sense to just use `mkNeovimPlugin` here.